EIPS/eip-7093.md

```diff
@@ -0,0 +1,409 @@
+
+   `permissions`= [`guardianA+signature`, `guardianB+signature`, ...]
+
+   The `permissions` is an array that consists of all the permissions of the Guardians who are participating in the recovery process.
+
+3. A bundler or another relayer service calls the `RecoveryAccount.startRecovery(0xA, newOwners, permissions)` function.
+
+4. `startRecovery()` function's processing logic is as follows:
+
+   - Generate a message hash (`msgHash`) from the input parameters `0xA`, `newOwners` and internally generated [EIP-712](./eip-712.md) signature parameters and `RecoveryNonce`.
+   - Extract `guardian` and corresponding `signature` from the input parameters `permissions` and process them as follows:
+     - If `guardianA.signer` is non-empty (Identity A), call `IGuardianPermissionVerifier(guardianVerifier1).isValidPermissions(signerA, msgHash, permissionA.signature)` to validate the signature.
+     - If `guardianA.signer` is empty (Identity B), call the internal function `SignatureChecker.isValidSignatureNow(guardianVerifier2, msgHash, permissionB.signature)` to validate the signature.
+
+5. After successful verification of all `guardians` signatures, fetch the associated `config` data for policyVerifier address `0xA` and call `IRecoveryPolicyVerifier(0xA).verifyRecoveryPolicy(guardians, config)`. The function `verifyRecoveryPolicy()` performs the following checks:
+
+   Note that the `guardians` parameter in the function refers to the guardians whose signatures have been successfully verified.
+
+   - Verify that `guardians` (Identity A and B) are present in `config.guardianInfos` list and are unique.
+   - Accumulate the `property` values of `guardians` (30 + 30 = 60).
+   - Compare the calculated result (60) with the `config.thresholdConfigs.threshold` ,the result is more than the first element (`threshold: 50, lockPeriod: 24 hours`) but less than the second element (`threshold: 100, lockPeriod: ""`), the validation is successful, and the lock period of 24 hours is returned.
+
+6. The `RecoveryAccount` saves a temporary state `{policyVerifier, newOwners, block.timestamp + 24 hours}` and increments `RecoveryNonce`. A `RecoveryStarted` event is emitted.
+
+7. After the expiry time, anyone (usually a relayer) can call `RecoveryAccount.executeRecovery()` to replace `newOwners`, remove the temporary state, complete the recovery, and emit a `RecoveryExecuteed` event.
+
+   
+
+## Rationale
+
+This standard aims to define interfaces for social recovery in smart contract accounts, enabling diverse Guardian types and flexible, customizable recovery policies. The goal is to ensure compatibility and security through a standardized recovery process.
+
+By abstracting identity verification and recovery policy verification interfaces, we decouple the verification logic from the social recovery process. This design allows the seamless addition of new identity verification methods or changes in recovery policies without altering the core recovery logic.
```

Comment:
This seems more like content for the motivation section. The rationale section should justify individual technical choices made within the EIP itself, while the motivation should justify the EIP as a whole. For example, you might want to explain why you use a recovery nonce in this section.